dts/arm/infineon/psoc6/psoc6_02/psoc6_02.dtsi

```diff
@@ -9,6 +9,110 @@
 
 / {
 	soc {
+		i2c0: i2c@40600000 {
+			compatible = "infineon,cat1-i2c";
+			reg = <0x40600000 0x10000>;
+			#address-cells = <1>;
+			#size-cells = <0>;
+			interrupts = <39 6>;
+			status = "disabled";
+		};
+		i2c1: i2c@40610000 {
+			compatible = "infineon,cat1-i2c";
+			reg = <0x40610000 0x10000>;
+			#address-cells = <1>;
+			#size-cells = <0>;
+			interrupts = <40 6>;
+			status = "disabled";
+		};
+		i2c2: i2c@40620000 {
+			compatible = "infineon,cat1-i2c";
+			reg = <0x40620000 0x10000>;
+			#address-cells = <1>;
+			#size-cells = <0>;
+			interrupts = <41 6>;
+			status = "disabled";
+		};
+		i2c3: i2c@40630000 {
+			compatible = "infineon,cat1-i2c";
+			reg = <0x40630000 0x10000>;
+			#address-cells = <1>;
+			#size-cells = <0>;
+			interrupts = <42 6>;
+			status = "disabled";
+		};
+		i2c4: i2c@40640000 {
+			compatible = "infineon,cat1-i2c";
+			reg = <0x40640000 0x10000>;
+			#address-cells = <1>;
+			#size-cells = <0>;
+			interrupts = <43 6>;
+			status = "disabled";
+		};
+		i2c5: i2c@40650000 {
+			compatible = "infineon,cat1-i2c";
+			reg = <0x40650000 0x10000>;
+			#address-cells = <1>;
+			#size-cells = <0>;
+			interrupts = <44 6>;
+			status = "disabled";
+		};
+		i2c6: i2c@40660000 {
+			compatible = "infineon,cat1-i2c";
+			reg = <0x40660000 0x10000>;
+			#address-cells = <1>;
+			#size-cells = <0>;
+			interrupts = <45 6>;
+			status = "disabled";
+		};
+		i2c7: i2c@40670000 {
+			compatible = "infineon,cat1-i2c";
+			reg = <0x40670000 0x10000>;
+			#address-cells = <1>;
+			#size-cells = <0>;
+			interrupts = <46 6>;
+			status = "disabled";
+		};
+		i2c8: i2c@40680000 {
+			compatible = "infineon,cat1-i2c";
+			reg = <0x40680000 0x10000>;
+			#address-cells = <1>;
+			#size-cells = <0>;
+			interrupts = <18 6>;
+			status = "disabled";
+		};
+		i2c9: i2c@40690000 {
+			compatible = "infineon,cat1-i2c";
+			reg = <0x40690000 0x10000>;
+			#address-cells = <1>;
+			#size-cells = <0>;
+			interrupts = <47 6>;
+			status = "disabled";
+		};
+		i2c10: i2c@406a0000 {
+			compatible = "infineon,cat1-i2c";
+			reg = <0x406a0000 0x10000>;
+			#address-cells = <1>;
+			#size-cells = <0>;
+			interrupts = <48 6>;
+			status = "disabled";
+		};
+		i2c11: i2c@406b0000 {
+			compatible = "infineon,cat1-i2c";
+			reg = <0x406b0000 0x10000>;
+			#address-cells = <1>;
+			#size-cells = <0>;
+			interrupts = <49 6>;
+			status = "disabled";
+		};
+		i2c12: i2c@406c0000 {
```

Comment:
There are 13 **blocks** on this part that can be I2C. They could also be a UART or SPI, if the user desires. They are called Serial Communication Blocks (SCB).

Reply:
Understood! I believe NXP's FlexComm/FlexIO, and RPI's PIO are similar in that regard and there is a bit of a discussion around those. It might be worth reviewing how those drivers and soc device tree definitions are laid out.

There is for example i2c_mcux_flexcomm and a ton of flexcomms on the rt685

The dtsi for that part is here:
https://github.com/zephyrproject-rtos/zephyr/blob/main/dts/arm/nxp/nxp_rt6xx_common.dtsi

And the flexcomms get used as i2c for example here:
https://github.com/zephyrproject-rtos/zephyr/blob/main/boards/arm/mimxrt685_evk/mimxrt685_evk_cm33.dts#L161

So you can split up the soc dtsi that defines your serial comm blocks and then instantiations of those comm blocks as i2c buses to the board devicetree.

Reply:
@teburd, thanks for the pointers. Given that we already have the UART version of this up with the pattern you highlighted, can we take this as an issue to fix in a later PR? We want to do the review and design of the NXP mechanism and how to apply it to our SoCs as its own task.

Reply:
That's fine, but I'd just reiterate doing this sooner rather than later is probably a good idea.